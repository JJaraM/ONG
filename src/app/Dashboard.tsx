import * as React from 'react';

import MaterialIcon, {colorPalette} from 'material-icons-react';
import {PullUpCard} from "../components/PullUpCard";

import {Table} from "../components/Table";
import TableRow from "../components/TableRow";
import TableColumn from "../components/TableColumn";
import localization from "./Localization";
import people from "../mock/people";


export class Dashboard extends React.Component<any, any> {

  constructor(props:any) {
    super(props);
  }



  createData = () => {
    const tableRows = Array<TableRow>();


    people.data.map((person, tagId) => {
      const tableColumns = Array<TableColumn>();

      const row = {} as TableRow;

      const id = {} as TableColumn;
      id.type = 'id';
      id.value = person.id;
      tableColumns.push(id);

      const avatar = {} as TableColumn;
      avatar.type = 'avatar';
      avatar.value = person.name;
      avatar.typeExtra = person.img;
      tableColumns.push(avatar);

      const project = {} as TableColumn;
      project.type = 'badge';
      project.className = 'badge badge-soft-danger badge-light';
      project.value = person.project;
      tableColumns.push(project);

      const action = {} as TableColumn;
      action.type = 'button';
      action.action = this.view;
      action.value = 'Action';
      tableColumns.push(action);

      row.columns = tableColumns;

      tableRows.push(row);
    })
    return tableRows;
  }

  view = (id: any) => {
    this.props.history.push("/pay/1");
  }

  render() {
    const tableRows = this.createData();
    const tableHeaders = [localization.common.contact, localization.common.project, localization.common.action];

    return (
        <div className="content-wrapper">

          <div className="pet-background">
            <div className="row p-b-60 p-t-60">
              <div className="col-md-10 mx-auto text-center text-white p-b-30">
                <div className="m-b-20">
                  <div className="avatar avatar-xl my-auto">
                      <img className="avatar-img rounded" src="https://o.aolcdn.com/images/dims3/GLOB/crop/4854x3182+0+0/resize/1028x675!/format/jpg/quality/85/http%3A%2F%2Fo.aolcdn.com%2Fhss%2Fstorage%2Fmidas%2Ff33fccca6149a8881f55cfbf6ac7ba1b%2F206112280%2Fgerman-shepherd-puppy-in-the-grass-picture-id841811318" alt=""/>
                  </div>
                  <h1>Fluxy</h1>
                </div>
              </div>
            </div>
          </div>


          <div className="content-wrapper pull-up">
            <div className="change-section row">
              <div className="col-lg-3 col-md-6 m-b-30">
                <PullUpCard message={localization.dashboard.pay} icon="arrow_upward" title="3 000 000" badge="badge-soft-success" />
              </div>
              <div className="col-lg-3 col-md-6 m-b-30">
                <PullUpCard message={localization.dashboard.pay} icon="arrow_upward" title="3 000 000" badge="badge-soft-success" />
              </div>
              <div className="col-lg-3 col-md-6 m-b-30">
                <PullUpCard message={localization.dashboard.pay} icon="arrow_upward" title="3 000 000" badge="badge-soft-success" />
              </div>
              <div className="col-lg-3 col-md-6 m-b-30">
                <PullUpCard message={localization.dashboard.pay} icon="arrow_upward" title="3 000 000" badge="badge-soft-success" />
              </div>
            </div>

          <div className="change-section row">
              {/*Start Table*/}
              <div className="col-lg-6 ">
                <Table title={"Pagos Pendientes"} headers={tableHeaders} rows={tableRows}/>
              </div>
              {/*End Table*/}

              {/*Start Table*/}
              <div className="col-lg-6 ">
                <Table title={"Desembolsos por aprobar"} headers={tableHeaders} rows={tableRows}/>
              </div>
              {/*End Table*/}
          </div>

          <div className="change-section row">
            <div className="col-lg-8 mx-auto  mt-2">
              <div className="card py-3 m-b-30">
                <div className="card-body">
                  <h3>Informacion de Fluxy</h3>
                  <p className="text-muted">
                    Use this page to update your contact information and change your password.
                  </p>
                  <div className="form-row">
                     <div className="form-group col-md-6">
                         <label>Name</label>
                         <input type="text" className="form-control" placeholder="Name"/>
                     </div>
                     <div className="form-group col-md-6">
                         <label>Email</label>
                         <input type="email" className="form-control" placeholder="Email" />
                     </div>
                     <button type="submit" className="btn btn-primary submit-big-button btn-tomato">Save changes</button>
                   </div>
                </div>
              </div>
            </div>
          </div>


        </div>
      </div>
    )
  }
}
