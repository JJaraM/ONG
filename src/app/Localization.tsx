import LocalizedStrings from 'react-localization';

const localization = new LocalizedStrings({
 en:{
   languages: {
     es: "Spanish",
     en: "English"
   },
   settings: {
     title: "Settings",
     theme_title: "Themes",
     theme_title_description: "Use the following section the app's color",
     change_theme: "Change",
     languages: "Languages",
     languages_description: "Use the following section to chage the app's language"
   },
   common: {
     contact: "Contact",
     action: "Action",
     project: "Project",
     save: "Save",
     saved: "Saved",
     saving: "Saving",
     back: "Back"
   },
   home: {
     title:"Register completly free",
   },
   dashboard: {
     pay: "Pay"
   },
   menu: {
     home: "Home",
     settings: "Settings",
     uiElements: "UI Elements",
     users: "Users",
     themes: "Themes"
   },
   user: {
     title: "Users",
     description: "View to show the user of the application",
     username: "Username",
     role: "Role(s)",
     permission: "Permission(s)",
     addUserTitle: "Add user",
     addUserDescription: "Use the following section to create an user",
     addRoleTitle: "Add Role",
     addRoleDescription: "Use the following section to create a role",
     addPermissionTitle: "Add Permission",
     addPermissionDescription: "Use the following section to create a permission",
     currentUsers: "Current Users",
     userOperations: "If you did not found the correct user, and you need to create a new one please go to",
     roleOperations: "If you did not found the correct role, and you need to create a new one please go to",
     permissionOperations: "If you did not found the correct permission, and you need to create a new one please go to"
   },
   pending_payment: {
     title: "Pending Payments",
     description: "View that displays the payments that needs to be approved",
     view: "View",
     approve: "Approve"
   }
 },
 es: {
   languages: {
     es: "Español",
     en: "Ingles"
   },
   settings: {
     title: "Configuraciones",
     theme_title: "Temas",
     theme_title_description: "Use la siguiente sección para cambiar los colores de la página",
     change_theme: "Cambiar",
     languages: "Lenguajes",
     languages_description: "Use la siguiente sección para cambiar el lenguaje de la aplicación"
   },
   common: {
     contact: "Contacto",
     action: "Acción",
     project: "Proyecto",
     save: "Guardar",
     saved: "Guardado",
     saving: "Guardando",
     back: "Atrás"
   },
   home: {
     title:"Regístrese completamente gratis",
   },
   dashboard: {
     pay: "Pagado"
   },
   menu: {
     home: "Inicio",
     settings: "Configuraciones",
     uiElements: "UI Elementos",
     users: "Usuarios",
     themes: "Temas"
   },
   user: {
     title: "Usuarios",
     description: "Vista para mostrar los usuarios de la aplicación",
     username: "Nombre de Usuario",
     role: "Role(s)",
     permission: "Permiso(s)",
     addUserTitle: "Agregar usuario",
     addUserDescription: "Use el siguiente formulario para agrear un usuario",
     addRoleTitle: "Agregar Role",
     addRoleDescription: "Use el siguiente formulario para agrear un role",
     addPermissionTitle: "Agregar Permiso",
     addPermissionDescription: "Use el siguiente formulario para agrear un permiso",
     currentUsers: "Usuarios Actuales",
     userOperations: "Si no encuentra el usuario necesario, y ocupa crear uno vaya al siguiente",
     roleOperations: "Si no encuentra el rol necesario, y ocupa crear uno vaya al siguiente",
     permissionOperations: "Si no encuentra el permiso necesario, y ocupa crear uno vaya al siguiente",
   },
   pending_payment: {
     title: "Pagos Pendientes",
     description: "Vista que muestra los pagos que necesitan ser aprobados",
     view: "Ver",
     approve: "Aprobar"
   }
 }
});

export default localization;
