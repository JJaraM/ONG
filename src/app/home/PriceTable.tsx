import * as React from 'react';

export class PriceTable extends React.Component<any, any> {

  constructor(props:any) {
    super(props);
  }

  render() {
    return (
      <section className="section-spacing">
        <div className="pricingTable">
          <h2 className="pricingTable-title">Encuentra el mejor plan para ti</h2>
          <h3 className="pricingTable-subtitle">Cada plan viene con una prueba gratuita de 1 mes</h3>
          <ul className="pricingTable-firstTable">
            <li className="pricingTable-firstTable_table">
              <h1 className="pricingTable-firstTable_table__header">Due√±o</h1>
              <p className="pricingTable-firstTable_table__pricing"><span>$</span><span>0</span><span>Month</span></p>
              <ul className="pricingTable-firstTable_table__options">
                <li>Mascotas ilimitadas</li>
                <li>Reporte de mascotas perdidas</li>
              </ul>
              <div className="pricingTable-firstTable_table__getstart">Get Started Now</div>
            </li><li className="pricingTable-firstTable_table">
              <h1 className="pricingTable-firstTable_table__header">Veterinario Gold</h1>
              <p className="pricingTable-firstTable_table__pricing"><span>$</span><span>19</span><span>Month</span></p>
              <ul className="pricingTable-firstTable_table__options">
                <li>Mascotas ilimitadas</li>
                <li>Manejo de expediente</li>
                <li>Manejo de expediente</li>
              </ul>
              <div className="pricingTable-firstTable_table__getstart">Get Started Now</div>
            </li><li className="pricingTable-firstTable_table">
              <h1 className="pricingTable-firstTable_table__header">Veterinario Platino</h1>
              <p className="pricingTable-firstTable_table__pricing"><span>$</span><span>49</span><span>Month</span></p>
              <ul className="pricingTable-firstTable_table__options">
                <li>Mascotas ilimitadas</li>
                <li>Manejo de expediente</li>
                <li>Manejo de expediente</li>
                <li>Factura electronico</li>
              </ul>
              <div className="pricingTable-firstTable_table__getstart">Get Started Now</div>
            </li>
          </ul>
        </div>
      </section>
    )
  }
}
