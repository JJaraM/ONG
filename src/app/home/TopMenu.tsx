import * as React from 'react';

import localization from "../Localization";

export class TopMenu extends React.Component<any, any> {

  constructor(props:any) {
    super(props);
  }

  render() {
    return (
      <header>
        <nav className="navbar navbar-expand-lg navbar-light">
          <div className="container">
            <a className="navbar-brand" href="">
              <img className="logo" src="img/logo.svg" alt="logo"/>
              <img className="logo-nav" src="img/logo-nav.svg" alt="logo-nav"/>
            </a>
            <div className="collapse navbar-collapse">
              <ul className="navbar-nav mr-auto">
                <li className="nav-item">
                  <a className="nav-link" href="#how-it-works">How it works
                    <span className="sr-only">(current)</span>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </nav>
      </header>
    )
  }
}
