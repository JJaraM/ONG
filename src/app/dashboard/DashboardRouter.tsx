import * as React from 'react';

import { TopMenu } from "./TopMenu";
import { SideMenu } from "./SideMenu";
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
//import { Router, Route, BrowserRouter, browserHistory } from 'react-router';
import { Dashboard } from "./Dashboard";
import { UIElements } from "./UIElements";
import { Settings } from "../settings/Settings";
import { UsersRouter } from "../settings/UsersRouter";
import { Themes } from "../settings/Themes";
import { JTheme } from '../../packages/theme/jTheme';
import { Language } from '../../packages/language/Language';
import "../../styles/dashboard.scss";
import UsersContainer from '../containers/UsersContainer';
import { Button, ToastContainer, toast } from 'mdbreact';
import * as Provider from 'react-redux';
import * as PropTypes from 'prop-types';
import { combineReducers, createStore } from 'redux';

const DashboardRouter = ({ store }) => (
    <Switch>
      <Route path="/settings/themes" exact component={Themes} />
      <Route exact path="/dashboard" component={Dashboard} />
      <Route path="/ui" exact component={UIElements} />
      <Route path="/settings/users" component={UsersContainer} />{/*UsersRouter*/}
      <Route path="/settings" exact component={() => <Settings event={this.reload} />}/>
    </Switch>
);

Language.setDefaultLanguage();
JTheme.changeTheme();

DashboardRouter.propTypes = {
  store: PropTypes.object.isRequired,
};

export default DashboardRouter;
