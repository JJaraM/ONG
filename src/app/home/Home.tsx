import * as React from 'react';

import "../../styles/home.scss";
import { PriceTable } from "./PriceTable";
import { TopMenu } from "./TopMenu";
import localization from "../Localization";

export class Home extends React.Component<any, any> {

  constructor(props:any) {
    super(props);
    this.setDefaultLanguage();
  }

  setDefaultLanguage() {
    localization.setLanguage('en');
  }

  signIn = () => {
    this.props.history.push("/dashboard");
  }

  render() {
    return (
      <>
        <TopMenu />
        <section className="section-spacing background-image header-section">
          <div className="container">
            <div className="row">
              <div className="col-md-6 col-lg-6">
                <div className="header-next-content">
                  <h1>
                    Lo mejor de las mascotas en un solo lugar
                  </h1>
                  <h2>
                    Perfecto para dueños y veterinarios
                  </h2>
                </div>
              </div>
              <div className="col-md-5 col-lg-5 offset-lg-1">
                <div className="form-content authentication-form">
                  <h3>

                      {localization.home.title}

                  </h3>
                  <div className="row">
                    <div className="form-group">
                      <input type="text" name="cta-name" className="form-control input-lg" placeholder="Nombre"/>
                    </div>
                    <div className="form-group">
                      <input type="text" name="cta-name" className="form-control input-lg" placeholder="Correo electronico"/>
                    </div>
                    <div className="form-group">
                      <input type="password" name="cta-name" className="form-control input-lg" placeholder="Contraseña"/>
                    </div>
                    <div className="form-group">
                      <div className="align-center-items">
                        <button className="btn btn-primary submit-big-button btn-tomato" onClick={this.signIn}>Registarse</button>
                      </div>
                    </div>
                    <p>
                      Al ingresar esta deacuerdo con nuestras terminos y politicas.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
        <PriceTable />
      </>
    )
  }
}
