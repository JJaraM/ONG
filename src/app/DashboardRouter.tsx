import * as React from 'react';

import { HashRouter, Route, Switch, Redirect} from "react-router-dom";
import { Dashboard } from "./Dashboard";
import { PetProfile } from "./PetProfile";
import { Pets } from "./Pets";
import { TopMenu } from "./dashboard/TopMenu";
import { SideMenu } from "./dashboard/SideMenu";
import { PayDetails } from "./pay/PayDetails";
import { UIElements } from "./UIElements";
import { Settings } from "./Settings";
import "../styles/dashboard.scss";
import "../styles/card.scss";
import themesData from "../styles/themes";

export class DashboardRouter extends React.Component<any, any> {

  constructor(props:any) {
    super(props);
  }

  componentDidMount() {
    this.setTheme();
  }

  setTheme = () => {
    let userConfigurationTheme = localStorage.getItem('theme');
    if (userConfigurationTheme === null || userConfigurationTheme === undefined || userConfigurationTheme === '') {
      userConfigurationTheme = 'masakali';
    }
    const theme = themesData.themes[userConfigurationTheme];
    Object.keys(theme).forEach((key) => {
      const cssKey = `--${key}`;
      const cssValue = theme[key];
      document.body.style.setProperty(cssKey, cssValue);
    });
  }


  render() {

    return (
      <div className="wrapper ">
        <TopMenu />
        <div className="container-fluid page-body-wrapper">
          <SideMenu />
          <div className="main-panel">
            <Switch>
              <Route path="/dashboard" exact component={Dashboard} />
              <Route path="/pets" exact component={Pets} />
              <Route path="/pay/:id" exact component={PayDetails} />
              <Route path="/pet/:id" exact component={PetProfile} />
              <Route path="/ui" exact component={UIElements} />
              <Route path="/settings" exact component={Settings} />
            </Switch>

          </div>
        </div>
      </div>
    );
  }
}
