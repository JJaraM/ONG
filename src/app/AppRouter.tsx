import * as React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { Home } from './home/Home';
import { DashboardRouter } from './DashboardRouter';

export class NoMatch extends React.Component{
    render(){
        return(
            <div>404 page</div>
        )
    }
}

class AppRouter extends React.Component {
  render() {
    return (
      <Router>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/dashboard" component={DashboardRouter} />
          <Route exact path="/ui" component={DashboardRouter} />
          <Route exact path="/settings" component={DashboardRouter} />
          <Route exact path="/pay/:id" component={DashboardRouter} />
          <Route id="noMatch" component={NoMatch} />
        </Switch>
      </Router>
    );
  }
}

export default AppRouter;
