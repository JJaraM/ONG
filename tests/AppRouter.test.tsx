import * as React from 'react';
import { BrowserRouter as Router} from 'react-router-dom';

import AppRouter, { NoMatch } from '../src/app/AppRouter';
import { DashboardRouter } from '../src/app/DashboardRouter';
import { Dashboard } from '../src/app/Dashboard';

import { Home, NoMatch} from '../src/app/home/Home';
import { MemoryRouter } from 'react-router';
import { shallow, mount } from 'enzyme';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

const renderRoutes = path => mount(
  <MemoryRouter initialEntries={[path]}>
    <AppRouter />
  </MemoryRouter>
);

describe('routes using memory router', () => {

  let element;

  beforeEach(() => {
    element = <AppRouter />
  });

  it('renders as expected', () => {
    const component = shallow(element);
    expect(component).toMatchSnapshot();
  });

  it('routes / to home', () => {
    const component = shallow(element);
    expect(component.find('Route[exact=true][path="/"]').first().prop('component')).toBe(Home);
  });

  it('routes /dashboard to DashboardRouter', () => {
    const component = shallow(element);
    expect(component.find('Route[exact=true][path="/dashboard"]').first().prop('component')).toBe(DashboardRouter);
  });

  it('routes /notMatch to notMatch', () => {
    const component = shallow(element);
    expect(component.find('Route[id="noMatch"]').first().prop('component')).toBe(NoMatch);
  });


});
