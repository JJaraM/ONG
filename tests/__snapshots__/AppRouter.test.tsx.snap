// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes using memory router renders as expected 1`] = `ShallowWrapper {}`;
